resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -493,7 +493,10 @@ private[spark] class Client(
           val localPath = getQualifiedLocalPath(localURI, hadoopConf)
           val linkname = targetDir.map(_ + "/").getOrElse("") +
             destName.orElse(Option(localURI.getFragment())).getOrElse(localPath.getName())
-          val destPath = copyFileToRemote(destDir, localPath, replication, symlinkCache)
+            var destPath = localPath
+          if (localPath.toUri.getScheme != "s3" && localPath.toUri.getScheme != "s3a") {
```

Comment:
maybe also  `if (localPath.toUri.getScheme.startsWith("s3") {` to accommodate even s3n scheme